sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/CatalystQlSuite.scala

```diff
@@ -170,4 +171,10 @@ class CatalystQlSuite extends PlanTest {
     parser.parsePlan("select sum(product + 1) over (partition by (product + (1)) order by 2) " +
       "from windowData")
   }
+
+  test("string charset literal") {
+    compareExpressions(
+      parser.parseExpression("'ISO-8859-1' 0x4341464562616265"),
```

Comment:
Talked to a few more people about this. I think it's best to just drop this. Only MySQL and Hive support this, and we cannot support the identical syntax anyway. I'd say if this is a really desired feature, we can just build a function for it.